src/category_theory/triangulated/rotate.lean

```diff
@@ -0,0 +1,496 @@
+as `Z[-1][1]` is not necessarily equal to `Z`, and `Z'[-1][1]` is not necessarily equal to `Z'`,
+but they are isomorphic, by the `counit_iso` of `shift C`)
+-/
+@[simps]
+def inv_rotate (f : triangle_morphism T₁ T₂) :
+  triangle_morphism (T₁.inv_rotate C) (T₂.inv_rotate C) :=
+{ hom₁ := f.hom₃⟦-1⟧',
+  hom₂ := f.hom₁,
+  hom₃ := f.hom₂,
+  comm₁' := begin
+    simp only [triangle.inv_rotate_mor₁],
+    rw [comp_neg, neg_comp, ← assoc],
+    dsimp,
+    rw [← functor.map_comp (shift C ).inverse, ← f.comm₃, functor.map_comp],
+    repeat {rw assoc},
+    suffices h : (shift C).unit_iso.inv.app T₁.obj₁ ≫ f.hom₁ =
+      (shift C).inverse.map ((shift C).functor.map f.hom₁) ≫ (shift C).unit_iso.inv.app T₂.obj₁,
+    { rw h },
+    { simp only [iso.hom_inv_id_app, assoc, equivalence.inv_fun_map,
+        nat_iso.cancel_nat_iso_inv_left],
+      exact (category.comp_id f.hom₁).symm }
+  end,
+  comm₃' := begin
+    have h := f.comm₂,
+    repeat {rw triangle.inv_rotate_mor₃},
+    rw [← assoc f.hom₂ _, ← f.comm₂],
+    dsimp,
+    repeat {rw assoc},
+    simp only [equivalence.fun_inv_map, iso.inv_hom_id_app_assoc],
+  end }
+
+end triangle_morphism
+
+/--
+Rotating triangles gives an endofunctor on the category of triangles in C.
+-/
+@[simps]
+def rotate : (triangle C) ⥤ (triangle C) :=
+{ obj := triangle.rotate C,
+  map := λ _ _ f, f.rotate C,
+  map_id' := begin
+    intro T₁,
+    simp only [triangle_category_to_category_struct_id],
+    ext,
+    { refl },
+    { refl },
+    { simp only [triangulated.triangle_morphism_id_hom₃,
+      triangulated.triangle_morphism_id_hom₁, triangle_morphism.rotate_hom₃],
+      dsimp,
+      rw (shift C).functor.map_id }
+  end,
+  map_comp' := begin
+    intros T₁ T₂ T₃ f g,
+    ext,
+    { refl },
+    { refl },
+    { dsimp,
+      rw (shift C).functor.map_comp }
+  end
+}
+
+/--
+The inverse rotation of triangles gives an endofunctor on the category of triangles in `C`.
+-/
+@[simps]
+def inv_rotate : (triangle C) ⥤ (triangle C) :=
+{ obj := triangle.inv_rotate C,
+  map := λ _ _ f, f.inv_rotate C,
+  map_id' := begin
+    intro T₁,
+    simp only [triangle_category_to_category_struct_id],
+    ext,
+    { simp only [triangle_morphism_id_hom₃, triangle_morphism.inv_rotate_hom₁,
+        triangle_morphism_id_hom₁],
+      dsimp,
+      rw (shift C).inverse.map_id },
+    { refl },
+    { refl }
+  end,
+  map_comp' := begin
+    intros T₁ T₂ T₃ f g,
+    ext,
+    { simp only [triangle_morphism.comp_hom₃, triangle_morphism.comp_hom₁, functor.map_comp,
+        triangle_morphism.inv_rotate_hom₁, triangle_category_to_category_struct_comp] },
+    { refl },
+    { refl }
+  end
+}
+
+/--
+There is a natural transformation between the identity functor on triangles,
+and the composition of a rotation with an inverse rotation.
+-/
+@[simps]
+def rot_comp_inv_rot_hom : 𝟭 (triangle C) ⟶ (rotate C) ⋙ (inv_rotate C) :=
+{ app := begin
+    intro T,
+    rw [functor.id_obj, functor.comp_obj],
```

Comment:
The `rw` isn't necessary here. In fact you don't even need to go into tactic mode. Just `app := λ T, { hom₁ := ... }`.

Reply:
(You'll need to update the proof of `comm₃'`.)